hazelcast-client/src/test/java/com/hazelcast/client/config/AbstractClientFailoverConfigBuilderTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.config;
+
+import com.hazelcast.core.HazelcastException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+abstract class AbstractClientFailoverConfigBuilderTest {
+    protected ClientFailoverConfig fullClientConfig;
+
+    @After
+    @Before
+    public void beforeAndAfter() {
+        System.clearProperty("hazelcast.client.failover.config");
+    }
+
+    @Test(expected = HazelcastException.class)
+    public void loadingThroughSystemProperty_nonExistingFile() {
+        System.setProperty("hazelcast.client.failover.config", "idontexist");
+        new YamlClientFailoverConfigBuilder();
```

Comment:
Why is the base abstract class using YamlClientFailoverConfigBuilder()? I would expect an abstract method instead.

Reply:
This is by mistake. Implemented in the XML and YAML test classes. Nice catch, thanks :+1: